admin/class-manage-fonts.php

```diff
@@ -229,6 +229,51 @@ function google_fonts_admin_page() {
 	<?php
 	}
 
+    function delete_font_asset ( $font_face ) {
+        // if the font asset is a theme asset, delete it
+        $theme_folder = get_stylesheet_directory();
+        $font_path = pathinfo( $font_face['src'][0] );
+        $font_dir = str_replace("file:./", "", $font_path['dirname']);
+
+        $font_asset_path = $theme_folder . DIRECTORY_SEPARATOR . $font_dir . DIRECTORY_SEPARATOR . $font_path['basename'];
+
+        if ( ! is_writable( $theme_folder . DIRECTORY_SEPARATOR . $font_dir ) ) {
+            return add_action( 'admin_notices', [ $this, 'admin_notice_font_asset_removal_error' ] );
+        }
+
+        if ( file_exists( $font_asset_path ) ) {
+            return unlink( $font_asset_path );
+        }
+        
+        return false;
+	}
+
+    protected function prepare_font_families_for_database( $font_families ) {
+		$prepared_font_families = array();
+
+		foreach ( $font_families as $font_family ) {
+			if ( isset ( $font_family['fontFace'] ) ) {
+				$new_font_faces = array();
+				foreach ( $font_family['fontFace'] as $font_face ) {
+					$updated_font_face = $font_face;
+					if ( !isset ( $font_face['shouldBeRemoved'] ) ) {
+						$new_font_faces[] = $updated_font_face;
+					} else {
+						$this->delete_font_asset( $font_face );
+					}
+				}
+
+				$font_family['fontFace'] = $new_font_faces;
+			}
+			if ( ! isset ( $font_family[ 'shouldBeRemoved' ] ) ) {
```

Comment:
Should this also contain the `else` logic to remove all of the fontFaces within the family?

Reply:
Thanks for the catch! I added the condition but on a previous `if` , to reuse the delete_font_face call. https://github.com/WordPress/create-block-theme/pull/176/commits/8692b1f12e6e3b2319805170a43089157f22c72b